test/js/metrics-api.test.js

```diff
@@ -2,10 +2,12 @@
 /* eslint-disable*/
 
 const fetch = require('node-fetch')
-import {logClick, logClickLink, logMenuChange, setMetricsApiMockFlag} from 'lib/metrics-api'
 import { screen, render, fireEvent } from '@testing-library/react'
 import React from 'react';
 
+import {logClick, logClickLink, logMenuChange, setMetricsApiMockFlag} from 'lib/metrics-api'
+import {shouldLog, setIsSuppressedEnv} from 'lib/sentry-logging'
```

Comment:
as mentioned above, let's replace this with mocking getSPContext().  Assuming we're going to expand the number of environment-specific things we do as time goes on, we'll want to become more adept at mocking the environment, rather than always having to build backdoor setters.